Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm

```diff
@@ -1427,6 +1427,23 @@ - (void)setInteractionState:(id)interactionState
     _page->restoreFromSessionState(sessionState, true);
 }
 
+- (BOOL)inspectable
+{
+#if ENABLE(REMOTE_INSPECTOR)
+    return _page->inspectable();
+#else
+    return NO;
+#endif
```

Comment:
wait so are we having `inspectable` be limited to remote inspection or not?  can we make this a full "line in the sand" moving away from remote inspection and instead be a global "can i be inspected", regardless of where the inspector is coming from?

Reply:
To an application using this framework/API on Cocoa platforms "remote inspection" is effectively equal to "inspection" because there is no supported API for in-process inspection. A global "inspectable" state is absolutely a long-term goal here, and the non-"remote" naming now will only help us later when we teach InspectorController how to use the "inspectable" property to determine if in-process inspection should be able to occur. That work is out of scope for this patch (but is an obvious "next step" when we have time).

So yes, we are trying to move away from the bifurcated idea in most places about remote v. in-process inspection, but this does not take us all the way there yet.

It is also worth noting that we will have to untangle at some point what is actually for remote inspection and what is for any inspection, since in most of JSC, remote inspection is the only inspection because there is no real way to do in-process inspection afaik, but JSC also has the shared inspection stuff that we need for both WebCore and JSC.

Reply:
ok if this is planned as a followup can we have a `// FIXME: ...`?

Reply:
Absolutely! I'll open a bugzilla to associate with the existing radar and try to hit key areas where the comment will be important.